agent/consul/state/txn.go

```diff
@@ -60,6 +60,12 @@ func (s *Store) txnKVS(tx WriteTxn, idx uint64, op *structs.TxnKVOp) (structs.Tx
 			err = fmt.Errorf("key %q doesn't exist", op.DirEnt.Key)
 		}
 
+	case api.KVGetNotExists:
+		_, entry, err = kvsGetTxn(tx, nil, op.DirEnt.Key, op.DirEnt.EnterpriseMeta)
+		if entry == nil && err == nil {
+			entry = &op.DirEnt
```

Comment:
The `api.KVSet` does a similar operation by returning the provided `op.DirEnt` to represent an empty key.

Reply:
🤔 I'm not sure I understand the comment here. In the case of `KVSet` `entry = &op.DirEntry` is just setting the entry to return to be the same one passed in but it should have a non-empty value. If the value is actually empty in the `op.DirEntry` then that means that after the op the key _will_ exist in the KV just with an explicitly empty value which is different to not existing. 

That said, the actual behaviour seems reasonable here, but it's not super obvious and perhaps a comment in the code to the effect of `return the passed `DirEntry` since it has the key set and a nil value as the response value`?

Reply:
Yeah, my comment was a little confusing -- sorry about that. Further down in the function, there's this little bit of code that clears out the value from the `KVSet` ops.

```
clone := entry.Clone()
clone.Value = nil
```

I was mainly pointing out that the `KVSet` still puts an entry in the results by re-using the `op.DirEnt` and nulling the value. That being said, I explicitly put in a `entry.Value = nil` into the code now so that it's guaranteed to always be nil, rather than returning back what the user specified, which is more clear.